cmd/kubepkg/templates/latest/deb/kubeadm/debian/control

```diff
@@ -10,6 +10,6 @@ Vcs-Browser: https://github.com/kubernetes/kubernetes
 
 Package: kubeadm
 Architecture: {{ .BuildArch }}
-Depends: kubelet (>= {{ index .Dependencies "kubelet" }}), kubectl (>= {{ index .Dependencies "kubectl" }}), kubernetes-cni (>= {{ index .Dependencies "kubernetes-cni" }}), cri-tools (>= {{ index .Dependencies "cri-tools" }}), ${misc:Depends}
+Depends: cri-tools (>= {{ index .Dependencies "cri-tools" }}), ${misc:Depends}
```

Comment:
I *suspect* that holding packaging will result in you not being able to install things because it is trying to pull in a dependency that it technically cannot install?  I can double-check this thought if you want me to.

I assume there are users who depend/assume that "kubeadm" upgrade == "kubelet and others" come along?

Reply:
AFAIK, it just holds the dependency but continues installing. please give it a try if you'd like.

> I assume there are users who depend/assume that "kubeadm" upgrade == "kubelet and others" come along?

there might be users that assume that a kubeadm package upgrade will trigger a kubelet (+other) package upgrades, yes.
that is because that is the current behavior.

if we can safely change it +1, but i think the following is a concern:
>> it will also apply to all patch versions of k8s, not only latest.

Reply:
This is the output of my test (skipped irrelevant outputs):

```
❯ docker run -it --rm ubuntu:focal
root@019275fc7b3c:/# apt-get update
root@019275fc7b3c:/# apt-get install -y apt-transport-https ca-certificates curl
root@019275fc7b3c:/# curl -fsSLo /usr/share/keyrings/kubernetes-archive-keyring.gpg https://packages.cloud.google.com/apt/doc/apt-key.gpg
root@019275fc7b3c:/# echo "deb [signed-by=/usr/share/keyrings/kubernetes-archive-keyring.gpg] https://apt.kubernetes.io/ kubernetes-xenial main" | tee /etc/apt/sources.list.d/kubernetes.list
root@019275fc7b3c:/# apt-get update
root@019275fc7b3c:/# apt-get install -y kube{let,adm,ctl}=1.19.16-00
root@019275fc7b3c:/# apt-mark hold kubelet kubeadm kubectl
root@019275fc7b3c:/# apt-get install kubeadm=1.20.5-00
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following held packages will be changed:
  kubeadm
The following packages will be upgraded:
  kubeadm
1 upgraded, 0 newly installed, 0 to remove and 3 not upgraded.
Need to get 6562 kB of archives.
After this operation, 131 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 https://packages.cloud.google.com/apt kubernetes-xenial/main arm64 kubeadm arm64 1.20.5-00 [6562 kB]
Fetched 6562 kB in 1s (6048 kB/s)  
debconf: delaying package configuration, since apt-utils is not installed
(Reading database ... 5305 files and directories currently installed.)
Preparing to unpack .../kubeadm_1.20.5-00_arm64.deb ...
Unpacking kubeadm (1.20.5-00) over (1.19.16-00) ...
Setting up kubeadm (1.20.5-00) ...
```

What's interesting is that it didn't even ask about the `kubelet` dependency.. I wonder if this was an issue before...?

Reply:
> What's interesting is that it didn't even ask about the kubelet dependency.. I wonder if this was an issue before...?

i don't think so.

Reply:
sounds good. I guess we'll have to wait and figure out where this comes from.